### DISCUSSION TITLE
Store $SYS messages in jetstream

### DISCUSSION THREAD

**Discussion by rafsawicki** [nats-io/nats-server#5046] [Q&A] [answered]
I'm trying to figure out how to store messages that are published as $SYS messages (more specifically, messages about MQTT clients connecting and disconnecting) in Jetstream, for at least once delivery and for audit purposes. So far, I've tried enabling Jetstream for system account (it's explicitly blocked) and using subject mapping to have them published as a regular message (mapping seem to be ignored). Is there any other way that I'm missing on how to achieve what I'm trying to do?

**Comment by ripienaar** [comment]
You need to create an account with jetstream enabled and create a stream there.

Then import as `stream` the advisory subjects into that account.

**Reply by wallyqs** [reply, marked as answer]
@rafsawicki you need to do it as a an import from another account that has JS enabled:
```sh
accounts {
  sys {
    users = [
      { user: mysys, pass: mysys }
    ]
    exports = [
      { stream: '$SYS.ACCOUNT.>' }
    ]
  }
  sysevents {
    jetstream = enabled
    users = [
      { user: myjs, pass: myjs }
    ]
    imports = [
      { stream { subject: '$SYS.ACCOUNT.>', account: sys } }
    ]
  }
}
```

**Reply by rafsawicki** [reply]
This works perfectly, thank you very much for providing a working example.
